FLEDGE.md

```diff
@@ -743,6 +743,66 @@ For detailed specification and explainers of the trusted key-value server, see a
 
 ##### 3.1.1 Cross-Origin Trusted Server Signals
 
+If the key-value server is on a different origin than the corresponding script, additional steps are
+needed to ensure that sensitive information is not inappropriately leaked to, or is manipulated by,
+a third party.
+
+In case of bidder signals, the following changes occur:
+1. The fetch is a cross-origin CORS fetch with `Origin:` set to the buyer script's origin.
+2. The value is passed to `crossOriginTrustedSignals` parameter, not the `trustedBiddingSignals`
+   parameter, and there is one more level of nesting denoting the server's origin, e.g:
+
+```
+{
+  'https://www.kv-server.example': {
+    'keys': {
+        'key1': arbitrary_json,
+        'key2': arbitrary_json,
+        ...},
+    'perInterestGroupData': {
+        'name1': {
+            'priorityVector': {
+                'signal1': number,
+                'signal2': number,
+                ...}
+        },
+        ...
+    }
+  }
+}
+```
+
+Seller signals have additional requirements, as the `trustedScoringSignalsURL` is provided by the
+publisher, not the seller:
+1. The seller script must provide an `Ad-Auction-Allow-Trusted-Scoring-Signals-From` response header,
+   a  [structured headers list of strings](https://www.rfc-editor.org/rfc/rfc8941) describing origins
+   fetching trusted signals from which is permitted. The trusted scoring signals fetch may not begin
+   until this header is received, in order to avoid leaking bid information to a third party.
+   This means using a cross-origin trusted server for seller information may carry a peformance
```

Comment:
>This means using a cross-origin trusted server for seller information may carry a peformance penalty.

I anticipate that a common seller use case will be a TSS URL that is _same-site_ with the scoring logic URL. Must this case incur the penalty?

Reply:
question: given that the original cross-origin fetch is CORS-based, how does `Access-Control-Max-Age` come into play here?  In other words, can we provide a mechanism to cache this granting of cross-site access?  As @dmdabbs correctly noted, the use case envisioned is a seller using different sub-domains from the same eTLD+1, and to avoid additional performance penalties as a result of this.

Reply:
Or, better yet, for those "cross-origin" sharing eLTD+1, is there a way to avoid this penalty entirely?

Reply:
Hmm, I need to figure out who to discuss this with. Your concern is eminently reasonable, but also generally it feels like web platform has been trying to be more origin than eTLD+1 based.

I think Access-Control-Max-Age doesn't come in at all since the signals fetch doesn't need a preflight, though I guess you're suggesting that the script headers sort of play the same role?

Reply:
I'm trying to better understand if we have pay this penlaty on each auction -- my comment about CORS was simply that we're borrowing those ACAO concepts.

Reply:
On today's WICG meeting, it was mentioned that perhaps the pre-connect may still begin, but it's the fetch() itself that is gated -- can we clarify?

Reply:
Preconnect should still work, only the load is gated, yes. (I should probably test that, though; this is based on staring at the code really hard).